File: pyround1/929.独特的电子邮件地址.py
#
# @lc app=leetcode.cn id=929 lang=python3
#
# [929] 独特的电子邮件地址
#

# @lc code=start
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        for i in range(len(emails)):
            for ch in emails[i]:
                at = emails[i].index("@")
                if ch == '.' and emails[i].index(ch) < at:
                    j = emails[i].index(ch)
                    emails[i] = emails[i][:j]+emails[i][j+1:]
                elif ch == '+':
                    k = emails[i].index(ch)
                    emails[i] = emails[i][:k]+emails[i][at:]
                    break
        return len(set(emails))


        
# @lc code=end

File: 929.独特的电子邮件地址.cpp
/*
 * @lc app=leetcode.cn id=929 lang=cpp
 *
 * [929] 独特的电子邮件地址
 */

// @lc code=start
class Solution {
public:
    int numUniqueEmails(vector<string>& emails) {
        unordered_map<string, int>cmp;
        for(auto str : emails){
            string s = "";
            int pos = 0;
            bool flag = false;
            while(pos < str.size()){
                if(str[pos] == '@')
                    flag = true;
                if(str[pos] == '.')
                {
                    if(flag)
                        s += str[pos];
                    pos++;
                }
                else if(str[pos] == '+'){
                    while(str[pos] != '@')
                        pos++;
                }
                else{
                    s += str[pos];
                    pos++;
                }
            }
            cout << s << "\n";
            cmp[s] += 1;
        }
        return cmp.size();
    }
};
// @lc code=end

